app/models/user.rb

```diff
@@ -90,12 +90,12 @@ class User < ActiveRecord::Base
   paginates_per 15
 
   # Relationships
-  has_and_belongs_to_many :roles
-  has_many :topics
-  has_many :posts
-  has_many :votes
-  has_many :docs
-  has_many :api_keys
+
+  has_many :topics, dependent: :delete_all
+  has_many :posts, dependent: :delete_all
```

Comment:
Yeah I agree.  Its tough to answer for all cases because if you are deleting spam or test users, then you would also want to destroy all dependent topics.  However, for a business, removing history is probably not what is desired.  I think it would be ideal to show a warning screen with a checkbox to allow the topics to be deleted with the user.